service/controller/v22/resource/bridgezone/resource.go

```diff
@@ -347,53 +345,97 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 			},
 			HostedZoneId: &intermediateZoneID,
 		}
-		time.Sleep(1 * time.Second)
 		_, err := defaultGuest.ChangeResourceRecordSetsWithContext(ctx, in)
 		if err != nil {
 			return microerror.Mask(err)
 		}
-		time.Sleep(1 * time.Second)
 
 		r.logger.LogCtx(ctx, "level", "debug", "message", "ensured deletion of final zone delegation from intermediate zone")
 	}
 
 	return nil
 }
 
+// findHostedZoneID fetches Route53 hosted zone IDs based on a given name. The
+// implementation fetches up to 100 matching results to find the right one. The
```

Comment:
I doubt it will take off that much pressure. We doubt we have more than 100 zones in any of tenant accounts. In the worst case scenario when all tenant zones and CP zones are in the same account the number of zones is `len(tenants) + 2` (2 for CP). There is potential of having multiple installations in the same account but then we have some many limits that I doubt we can create >100 clusters in the account. Even if we do there will be one on at very best two requests less per loop.

This is not to say that the change isn't good, but I don't know if we can happily drop those `time.Sleep` calls. It would be good to handle limits somehow instead. E.g. not running the resource for some time if we are close to hit the limit.

Reply:
There was a PM a while ago because the API rate limit kicked in and our CI builds failed. We have an account in which we do all kinds of things and start dozens of clusters at the same time for CI reasons. I want to get rid of the sleeps. Thus the implementation has to change.

Reply:
Sure. I just doubt this is a remedy for that, because it doesn't really reduce the number of requests (unless I'm missing something).

Reply:
As I said, the amount of records we have in some accounts is the difference, where the improved implementation is beneficial.

Reply:
I just checked. We have currently 8 zones in CP test AWS account, and 5 zones in tenant test AWS account. My guess would be that it is e2e clusters that create the pressure (i.e. number of running bridgezone resources), not the number of zones. Having separate test account could solve the issue.

Reply:
Yes, it is the number of e2e tests. We do not have separate test accounts and and nobody is going to sort this out soon.